src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -0,0 +1,943 @@
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X {
+    public int i;
+    public X(int i) => this.i = i;
+    public static void Main()
+    {
+        X x = new(42);
+        Console.Write(x.i);
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "42");
+        }
+
+        [Fact]
+        public void TestSafeCast()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X 
+{
+    public static void Main()
+    {
+        Console.Write(new() as X);
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (7,23): error CS9365: The first operand of an 'as' operator may not be a target-typed 'new'.
+                //         Console.Write(new() as X);
+                Diagnostic(ErrorCode.ERR_TypelessNewInAs, "new() as X").WithLocation(7, 23)
+                );
+        }
+
+        [Fact]
+        public void TestTupleElement()
+        {
+            var comp = CreateCompilation(@"
+using System;
+
+class X {
+    static void M((X a, X b) t) => Console.Write($""{t.a}{t.b}"");
+
+    public static void Main()
+    {
+        M((new(), new()));
+    }
+}
+" + trivial2uple + tupleattributes_cs, options: TestOptions.ReleaseExe).VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "XX");
+        }
+
+        [Fact]
+        public void TestConst()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class X {
+    public static void Main()
+    {
+        const int x = new();
+        Console.Write(x);
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (6,23): error CS9366: The default constructor of the value type 'int' may not be used with target-typed 'new'. Consider using 'default' instead.
+                //         const int x = new();
+                Diagnostic(ErrorCode.ERR_DefaultValueTypeCtorInTargetTypedNew, "new()").WithArguments("int").WithLocation(6, 23)
+                );
+        }
+
+        [Fact]
+        public void TestBadTargetType_Assignment()
+        {
+            var comp = CreateCompilation(@"
+using System;
+struct Struct {}
+abstract class Abstract {}
+static class Static {}
+interface Interface {}
+enum Enumeration {}
+unsafe class C
+{
+    public void Test<T>()
+    {
+        var v0 = new();
+        Struct v1 = new();
+        Action v2 = new();
+        Static v3 = new();
+        Abstract v4 = new();
+        Interface v5 = new();
+        Enumeration v6 = new();
+        int v7 = new();
+        int* v8 = new();
+        int? v9 = new();
+        (int, int) v10 = new();
+        dynamic v11 = new();
+        int[] v12 = new();
+        Error v13 = new();
+        T v14 = new();
+        ValueTuple<int, int> v15 = new();
+        ValueTuple<int, int> v16 = new(2, 3);
+    }
+}
+" + trivial2uple, options: TestOptions.UnsafeReleaseDll.WithWarningLevel(0)).VerifyDiagnostics(
```

Comment:
>WithWarningLevel(0) [](start = 56, length = 19)

Why was the warning level change needed here?

Reply:
Looks like UnsafeReleaseDll does contain tuple definitions but ReleeaseExe does not. I'll revert.